static/src/js/components/KeywordTree/KeywordTree.jsx

```diff
@@ -225,18 +235,80 @@ const KeywordTreeComponent = forwardRef(({
     }
   ]
 
-  const handleDelete = (nodeId) => {
-    setTreeData((prevData) => {
-      const deleteNode = (nodes) => nodes.filter((node) => node.id !== nodeId)
-        .map((node) => ({
-          ...node,
-          children: node.children ? deleteNode(node.children) : undefined
-        }))
+  const closeDeleteModal = () => {
+    setShowDeleteConfirmation(false)
+    setNodeToDelete(null)
+    setDeleteError(null)
+  }
+
+  const handleDeleteConfirmation = async () => {
```

Comment:
I think there is too much logic in the client.   The client really should just send the request to KMS to delete the concept, then refresh the tree, if successful -- KMS should do the check to ensure it is a leaf node and if not, throw the error.

Reply:
Done

Reply:
'Delete' only shows up when leaf. The tree will be refreshed after deleting a leaf and open the parent.